onyo/lib/onyo.py

```diff
@@ -1352,3 +1358,32 @@ def unset(self, paths: Iterable[Union[Path, str]], keys: list[str],
             self.restore()
 
         return diff
+
+    def get(
+            self, keys: Set[str], paths: Set[Path], depth: int,
+            filters: list[Filter]) -> Generator:
+        """
+        Get keys from assets matching paths and filters.
+        """
+        # filter assets by path and depth relative to paths
+        assets = [
```

Comment:
I think this should use the set_sanitize() function that set() and unset() both use, too, since that controlls the depth and everything, and will make it way easier to keep them all behaving the same :)

Reply:
I've looked into this and it's not needed because of the restricted functionality of `Repo.get()`. I use `self.assets` to retrieve all assets in the repository and then filter them based on the given paths and depth. I plan to add a warning to `commands.get()` when a protected or non-existing path is used, but crucially like `ls` the command should still return the output for the valid paths.

The more I think about this, the more I convince myself that we should extend `pathlib.Path` with an `OnyoPath` class where you can call properties like `OnyoPath.protected` that returns a bool. That way we can simply apply a filter to remove protected paths, existing paths, and other problematic cases.

Reply:
The idea about the `OnyoPath` I mention in the beginning of #325.

I can see, that probably not all functionality of `set_sanitize()` is needed, but I want to state clearly again, that I do not think `onyo get` should be merged as long as the `--depth` is different from the ones of `set` and `unset.`
That is why I think, needed or not, they all should use the same function for now to solve this in the same way.

Reply:
Now I'm confused. I've adapted the tests and function to use `--depth 0` as a no-limit default, which doesn't work with `set_sanitize`. I realized this just now, because `set_sanitize` doesn't just test paths and depth for validity, it also retrieves assets. If I am to use this prior to the PR being accepted, I will have to undo the changes to `--depth 0` as a no-limit default.

How about instead we address this issue in a PR and merge it in before this one?

Reply:
I am happy either way and do not have strong feelings about how it is resolved. :)
The only thing that matters for this flag is in my opinion consistency! :)

Reply:
With the acceptance of PR #329 the get, set, and unset functions now use the same asset-retrieval approach and this issue has been addressed.